src/Components/Pages/TransferPage.tsx

```diff
@@ -750,20 +758,36 @@ const TransferPage = () => {
                     </section>
                   </section>
                   <section>
-                    <AddressInput
-                      disabled={!destinationChainConfig}
-                      name="receiver"
-                      label="Destination Address"
-                      placeholder="Please enter the receiving address..."
-                      className={classes.address}
-                      classNames={{
-                        input: classes.addressInput,
-                      }}
-                      senderAddress={`${address}`}
-                      sendToSameAccountHelper={
-                        destinationChainConfig?.type === homeConfig?.type
-                      }
-                    />
+                      {!isPasteDestinationAddressManually &&
+                      isDestinationBridgeAddressList ? (
+                          <AddressSelectInput
+                              name="receiver"
+                              addresses={destinationBridge.addresses}
+                          />
+                      ): (
+                          <AddressInput
+                              disabled={!destinationChainConfig}
+                              name="receiver"
+                              label="Destination Address"
+                              placeholder="Please enter the receiving address..."
+                              className={classes.address}
+                              classNames={{
+                                  input: classes.addressInput,
+                              }}
+                              senderAddress={`${address}`}
+                              sendToSameAccountHelper={
+                                  destinationChainConfig?.type === homeConfig?.type
+                              }
+                          />
+                      )}
+                      {isDestinationBridgeAddressList && (
+                          <CheckboxInput
+                              className={classes.destinationAddressCheckbox}
+                              label="Paste Address Manually"
+                              value={isPasteDestinationAddressManually}
+                              onChange={() => setIsPasteDestinationAddressManually(!isPasteDestinationAddressManually)}
+                          />
+                      )}
```

Comment:
The logic is a bit weird. Can we use a component that can be used as an "input" and "select" at the same time

Reply:
I made this decision based on two primary reasons:
1. The UI kit does not currently offer a component that allows for both selection and input simultaneously.
2. While we use Whitelisting (via the Polkadot-JS Web Extension) for transfers from EVM to Substrate, transfers from Substrate to EVM still require manual input.

Reply:
Point two can usually be handled by such a component but if the UI kit doesn't have it let's keep your solution